### PR TITLE
Move to C++20 and remove vector_utils_legacy

### PR DESCRIPTION
**Pull request by jmcarcell**
for the next release of the Key4hep stack we may want to consider moving to C++20 so that C++20 features can be used without automatically introducing incompatibilities, and they are easier to adopt if we are already using C++20. This would help make the release usable for a longer time. The Key4hep stack has been built with C++20 for many months by now. There are the LCG stacks on C++17 as far as I know and it's possible other people are using, for example, an older version of ROOT that can't be built with C++20 in which case this may have to wait.

BEGINRELEASENOTES
- Move to C++20 and remove vector_utils_legacy

ENDRELEASENOTES

### CODE REVIEW

File: .github/workflows/lcg_linux_with_podio.yml [hunk 1]
```diff
@@ -12,11 +12,6 @@ jobs:
               "dev4/x86_64-el9-clang16-opt",
               "dev4/x86_64-el9-gcc13-opt"]
         CXX_STANDARD: [20]
-        include:
-          - LCG: "LCG_106/x86_64-el9-gcc13-opt"
```
[the remark this answers is not in the document]
Comment (andresailer) [reply .github/workflows/lcg_linux_with_podio.yml:16]: gcc13 is build with c++20 (so is clang16). Nothing on ubuntu yet.

[the remark this answers is not in the document]
Comment (tmadlener) [reply .github/workflows/lcg_linux_with_podio.yml:16]: I think gcc11 on ubuntu22.04 should have enough c++20 capabilities for EDM4hep.

**Comment by tmadlener** [comment]
Looks like LCG 106 on Ubuntu 22.04 does not have a c++20 capable ROOT yet? https://github.com/key4hep/EDM4hep/actions/runs/10486852425/job/29045992210?pr=343#step:4:751

**Comment by andresailer** [comment]
Only LCG stacks with gcc13 and clang16 or newer have c++20 enabled.

**Comment by jmcarcell** [comment]
Ubuntu removed, can be added back if it has C++20, anyway the Key4hep builds have Ubuntu.

**Pull request merged**